spec/javascripts/health-care/components/form-elements/GrowableTable.spec.jsx

```diff
@@ -0,0 +1,174 @@
+
+import GrowableTable from '../../../../../_health-care/_js/components/form-elements/GrowableTable';
+
+class Row extends React.Component {
+  render() {
+    return (
+      <div/>
+    );
+  }
+}
+
+describe('<GrowableTable>', () => {
+  describe('propTypes', () => {
+    let consoleStub;
+    beforeEach(() => {
+      consoleStub = sinon.stub(console, 'error');
+    });
+
+    afterEach(() => {
+      consoleStub.restore();
+    });
+
+    /** Mock function used to verify propType checks for functions. */
+    const func = () => {};
+
+    it('component is required', () => {
+      SkinDeep.shallowRender(
+        <GrowableTable rows={[]} createRow={func} onRowsUpdate={func}/>);
+      sinon.assert.calledWithMatch(consoleStub, /Required prop `component` was not specified in `GrowableTable`/);
+    });
+
+    it('component must be a func', () => {
+      SkinDeep.shallowRender(
+        <GrowableTable rows={[]} component createRow={func} onRowsUpdate={func}/>);
+      sinon.assert.calledWithMatch(consoleStub, /Invalid prop `component` of type `boolean` supplied to `GrowableTable`, expected `function`/);
+    });
+
+    it('createRow is required', () => {
+      SkinDeep.shallowRender(
+        <GrowableTable rows={[]} component={func} onRowsUpdate={func}/>);
+      sinon.assert.calledWithMatch(consoleStub, /Required prop `createRow` was not specified in `GrowableTable`/);
+    });
+
+    it('createRow must be a func', () => {
+      SkinDeep.shallowRender(
+        <GrowableTable rows={[]} component={func} createRow onRowsUpdate={func}/>);
+      sinon.assert.calledWithMatch(consoleStub, /Invalid prop `createRow` of type `boolean` supplied to `GrowableTable`, expected `function`/);
+    });
+
+    it('onRowsUpdate is required', () => {
+      SkinDeep.shallowRender(
+        <GrowableTable rows={[]} component={func} createRow={func}/>);
+      sinon.assert.calledWithMatch(consoleStub, /Required prop `onRowsUpdate` was not specified in `GrowableTable`/);
+    });
+
+    it('onRowsUpdate must be a func', () => {
+      SkinDeep.shallowRender(
+        <GrowableTable rows={[]} component={func} createRow={func} onRowsUpdate/>);
+      sinon.assert.calledWithMatch(consoleStub, /Invalid prop `onRowsUpdate` of type `boolean` supplied to `GrowableTable`, expected `function`/);
+    });
+
+    xit('rows is required', () => {
+      SkinDeep.shallowRender(
+        <GrowableTable component={func} createRow={func} onRowsUpdate={func}/>);
+      sinon.assert.calledWithMatch(consoleStub, /Required prop `rows` was not specified in `GrowableTable`/);
+    });
+
+    xit('rows must be an array', () => {
+      SkinDeep.shallowRender(
+        <GrowableTable rows component={func} createRow={func} onRowsUpdate={func}/>);
+      sinon.assert.calledWithMatch(consoleStub, /Required prop `rows` was not specified in `GrowableTable`/);
+    });
+  });
+
+  describe('row management', () => {
+    let growableTable;
+    let currentId = 0;
```

Comment:
What if this is initialized to `2` and then the rows have values `0` and `1` instead of `v1` and `v2`? It took me a little while to read through this and understand how these things were related.